src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -2373,46 +2364,63 @@ class StubGenerator: public StubCodeGenerator {
              tmpU = isLU ? tmp5 : tmp1, // where to keep U for comparison
              tmpL = isLU ? tmp1 : tmp5; // where to keep L for comparison
 
-    __ sub(tmp2, strL, cnt2); // strL pointer to load from
-    __ slli(t0, cnt2, 1);
-    __ sub(cnt1, strU, t0); // strU pointer to load from
+    // make sure main loop is 8 byte-aligned, we should load another 4 bytes from strL
+    __ beqz(cnt2, DONE);  // no characters left
```

Comment:
I don't quite understand this newly-added branch. If the intention was to ensure that we have at lease 4 chars left, shoudn't we compare with 4 instead? But given that the const `STUB_THRESHOLD` is (64 + 8) in C2_MacroAssembler::string_compare, I think neigher will be taken.

Reply:
you right, that looks to be a sanity check where the branch will never be taken, will remove both